app/features/quizzify/loaders.py

```diff
@@ -0,0 +1,227 @@
+
+    
+    def loadTXT(self, byte_file, index):
+        file = self.files[index]
+        startLine = file.startLine or 1
+        endLine = file.endLine or 10**9
+        content = []
+        docs = []
+        for current_line_number, line in enumerate(byte_file, start=1):
+            if startLine <= current_line_number <= endLine:
+                text = line.decode('utf-8')
+                content.append(text)
+            elif current_line_number > endLine:
+                break
+        result_text = ' '.join(s.strip() for s in content)
+        metadata = {"source": "txt"}
+        doc = Document(page_content=result_text, metadata=metadata)
+        docs.append(doc)
+        return docs
+
+
+
+class WebPageLoader:
+    def __init__(self, verbose):
+        self.verbose = verbose
+
+    def load(self, files) -> List[Document]:
+        docs = []
+        for file in files:
+            try:
+                url = file.url
+                tag = file.tag
+                response = requests.get(url)
+                soup = BeautifulSoup(response.content,'html.parser')
+                if tag:
+                    content = soup.find_all(tag)
+                    text = ' '.join(data.get_text().strip() for data in content)
+                else:
+                    text = soup.get_text(separator="\n", strip=True)
+
+                metadata = {"source": "webpage"}
+                doc = Document(page_content=text, metadata=metadata)
+                docs.append(doc)
+
+            except Exception as e:
+                logger.error(f"An error occurred while processing web page at {file}: {str(e)}")
+                raise LoaderError(f"Error loading web page: {file}") from e
+
+        return docs
+
+
+class CustomYoutubeLoader:
+
+    def __init__(self, verbose=None):
+        self.verbose = verbose
+
+    def transcript_check(self, video_id):
+        try:
+            transcript = YouTubeTranscriptApi.get_transcript(video_id=video_id,
+                                                             languages=['en', 'en-GB', 'en-US', 'en-AU', 'en-BZ',
+                                                                        'en-CA', 'en-IE', 'en-JM', 'en-NZ', 'en-ZA',
+                                                                        'en-TT', 'en-GB'])
+        except:
+            return False
+
+        return transcript
+
+
+    def load(self, files):
+
+        documents = []
+        for file in files:
+            url = file.url
+            start_timestamp = file.start_timestamp
+            end_timestamp = file.end_timestamp 
+
+            video_id = YoutubeLoader.extract_video_id(url)
+
+            data = ''
+
+            transcript_data = self.transcript_check(video_id)
+            if transcript_data:
+                try:
+                    
+                    # Filter transcript based on start_time and end_time
+                    if start_timestamp is not None and end_timestamp is not None:
```

Comment:
This is hardcoded. Please, be careful with these conditionals as they could be segregated using validation functions.